packages/@aws-cdk/aws-lambda-python/lib/Dockerfile.dependencies

```diff
@@ -0,0 +1,18 @@
+# The correct AWS SAM build image based on the runtime of the function will be
```

Comment:
Only other questions I had here were,
1. Why are we using `rsync` instead of `cp`? Will we ever copy from a URL? If not, is `cp -r` sufficient?
2. The current sam image has caching disabled due to user permissions. Is there a way to enable it with just image args?

If `cp` is sufficient, and we can enable pip caching, then could we do away with these two Dockerfiles?

I suppose pipenv support is another reason to keep them. I just wanted to make sure these artifacts were absolutely necessary.
If `cp`

Reply:
> 1. Why are we using `rsync` instead of `cp`? Will we ever copy from a URL? If not, is `cp -r` sufficient?
I'm unsure as to why.  Another PR #9763 added this after I started. Perhaps @adamelmore can answer your question.

> 2. The current sam image has caching disabled due to user permissions. At least that's what I've experienced in practice. Is there a way to enable it with just image args?
> 
> If `cp` is sufficient, and we can enable pip caching, then could we do away with these two Dockerfiles?
> 
> I suppose pipenv support is another reason to keep them. I just wanted to make sure these artifacts were absolutely necessary.

It may be possible to expose a persistent location for the pip cache through a volume and dedicated directory in the user's project directory. But, Docker volume performance can be quite poor in macOS and Windows. Ultimately, the Dockerfiles were used to bake-in the pip dependencies to improve performance by only re-running pip when a change is detected in `requirements.txt` or the `Pipfile` or `Pipfile.lock`. For these reasons, I'd tend to prefer the Dockerfile approach.